src/examples/unofficial/csharp.md

```diff
@@ -0,0 +1,128 @@
+# Using C#
+
+Added by [Sylveon](https://github.com/Sylveon76)
+
+## Getting started
+
+### Installing the package
+
+Add the NuGet package `NekosBestApiNet` to your project:
+
+```ps
+dotnet add package NekosBestApiNet
+```
+
+### Simple usage
+
+
+```cs
+using System;
+using System.Threading.Tasks;
+using NekosBestApiNet;
+
+public class ExampleClass
+{
+  private readonly NekosBestApi _nekosBestApi;
+
+  public ExampleClass() 
+    => _nekosBestApi = new NekosBestApi();
+
+  public async Task Hug() 
+  {
+    var image = await _nekosBestApi.ActionsApi.Hug();
+
+    Console.WriteLine(image.Results.FirstOrDefault().Url);
+  }
+}
+```
+
+### Example using custom HttpClient
+
+You can provide your own HttpClient instance, but you have to set the BaseAddress manually
+
+```cs
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using NekosBestApiNet;
+
+public ExampleClass
+{
+  var httpClient = new HttpClient 
+  {
+    BaseAddress = new Uri("https://nekos.best/api/v2")
+  };
+
+  _nekosBestApi = new NekosBestApi(httpClient);
+}
+```
+
+### Example using dependency injection
+
+Create a ServiceCollection, then add an instance of the NekosBestApi class to it
+
+
+```cs
+using System;
+using System.Threading.Tasks;
+using NekosBestApiNet;
+using Microsoft.Extensions.DependencyInjection;
```

Comment:
Perhaps you should mention that this assembly reference is provided by the package `Microsoft.Extensions.DependencyInjection`

Reply:
I'm pretty sure they would already know this if using dependency injection